lib/elixir/pages/Guards.md

```diff
@@ -0,0 +1,226 @@
+# Guards
+
+Guards are a way to augment pattern matching in a predefined set of constructs.
+
+Let's see an example of a guard used in a function clause:
+
+
+```elixir
+def long_list?(list) when length(list) > 3,  do: true
```

Comment:
I think `map_size/1` check would be better here, and it will be educational (shows how to check 0-size map). :bowtie: